src/Service.Tests/Caching/HealthEndpointCachingTests.cs

```diff
@@ -0,0 +1,186 @@
+
+            // Simulate a delay to allow the cache to expire
+            // and force a new request to be sent to the database.
+            Task.Delay((cacheTtlSeconds ?? EntityCacheOptions.DEFAULT_TTL_SECONDS) * 1000 + 1000).Wait();
+
+            HttpRequestMessage healthRequest2 = new(HttpMethod.Get, "/health");
+            response = await client.SendAsync(healthRequest2);
+            string responseContent2 = await response.Content.ReadAsStringAsync();
+
+            // Regex to remove the "timestamp" field from the JSON string
+            string pattern = @"""timestamp""\s*:\s*""[^""]*""\s*,?";
+            responseContent1 = Regex.Replace(responseContent1, pattern, string.Empty);
+            responseContent2 = Regex.Replace(responseContent2, pattern, string.Empty);
+
+            Assert.AreNotEqual(responseContent2, responseContent1);
+        }
+    }
+
+    /// <summary>
+    /// Simulates GET request to DAB's comprehensive health check endpoint ('/health') and validates the contents of the response.
+    /// The expected behavior is that both these response should be same in case cache is enabled with no delay.
+    /// </summary>
+    [TestMethod]
+    [TestCategory(TestCategory.MSSQL)]
+    [DataRow(null, DisplayName = "Validation of default cache TTL and no delay.")]
+    [DataRow(0, DisplayName = "Validation of cache TTL set to 0 and no delay.")]
+    [DataRow(10, DisplayName = "Validation of cache TTL set to 10 and no delay.")]
+    public async Task ComprehensiveHealthEndpointCachingValidateNoDelay(int? cacheTtlSeconds)
+    {
+        Entity requiredEntity = new(
+            Health: new(Enabled: true),
+            Source: new("books", EntitySourceType.Table, null, null),
+            Rest: new(Enabled: true),
+            GraphQL: new("book", "books", true),
+            Permissions: new[] { ConfigurationTests.GetMinimalPermissionConfig(AuthorizationResolver.ROLE_ANONYMOUS) },
+            Relationships: null,
+            Mappings: null);
+
+        Dictionary<string, Entity> entityMap = new()
+        {
+            { "Book", requiredEntity }
+        };
+
+        CreateCustomConfigFile(entityMap, cacheTtlSeconds);
+
+        string[] args = new[]
+        {
+            $"--ConfigFileName={CUSTOM_CONFIG_FILENAME}"
+        };
+
+        using (TestServer server = new(Program.CreateWebHostBuilder(args)))
+        using (HttpClient client = server.CreateClient())
+        {
+            HttpRequestMessage healthRequest1 = new(HttpMethod.Get, "/health");
+            HttpResponseMessage response = await client.SendAsync(healthRequest1);
+            string responseContent1 = await response.Content.ReadAsStringAsync();
+            Assert.AreEqual(expected: HttpStatusCode.OK, actual: response.StatusCode, message: "Received unexpected HTTP code from health check endpoint.");
+
+            // No delay, force a new request to be sent to the database.
```

Comment:
```suggestion
            // No delay, so we won't force a new request to be sent to the database.
```

Shouldn't this be we won't force since there is no delay?

Reply:
Modified the comment to 
```
// Simulate no delay scenario to make sure that the cache is not expired (in case available)
// and force a new request to be sent to the database.
// Further check if the responses are same.
```
 Does this make more sens?